### DISCUSSION TITLE
Resolution of WixVariables defined in the wixlib

### DISCUSSION THREAD

**Discussion by drolevar** [wixtoolset/issues#6373] [Questions] [answered]
As far as I understand, wix3 was not resolving WixVariables when building a wixlib. As a result, the corresponding table row in the wixlib looked as follows:
```
<row sourceLineNumber="c:\Projects\wix3\src\ext\BalExtension\wixlib\wixstdba.wxs*43">
	<field>pay59BE46628EB74147621677BDE0987BF7</field>
	<field>thm.xml</field>
	<field cabinetFileId="42">!(wix.WixStdbaThemeXml=HyperlinkTheme.xml)</field>
	<field/>
	<field>1</field>
	<field>!(wix.WixStdbaThemeXml=HyperlinkTheme.xml)</field>
	<field>1</field>
</row>
```
In the wix4 the paths are resolved by the `WixToolset::Core::Librarian::ResolveFilePathsToEmbed` method, so the corresponding part in the new wixlib looks like that:
```
"type": "WixBundlePayload",
"id": { "id": "payViPRdlZD7MPIBgwmX2Hv5C6oKt4", "access": "section" },
"ln": { "file": "c:\\Projects\\wix4\\Bal.wixext\\src\\wixlib\\wixstdba.wxs", "line": 9 },
"fields": [
	{
		"data": "thm.wxl"
	},
	{
		"context": "wix.lib",
		"data": { "embed": "true", "path": "wix-ir/RtfTheme.wxl" },
		"prev": { "data": { "path": "!(wix.WixStdbaThemeWxl=RtfTheme.wxl)" } }
	},
	null,
	{
		"data": true
	},
	{
		"data": "!(wix.WixStdbaThemeWxl=RtfTheme.wxl)"
	},
        <...>
]
```
So the SourceFile field becomes fully resolved and then if the WixVariable is overridden elsewhere it doesn't affect the file which is being used.
In wix3 the librarian was collecting the list of files to be embedded in a bundle, but here they are being evaluated in-place.
I wonder if it's a bug or am I misunderstanding something.

**Comment by rseanhall** [comment]
If the file specified in `bal:WixStandardBootstrapperApplication/@ThemeFile` is not getting used in the bundle as the theme file, then that's a bug. If you are interested in discussing how to fix that bug, then that belongs on the [wix-devs mailing list](https://wixtoolset.org/documentation/mailinglist).

**Comment by robmen** [comment, marked as answer]
This sounds like a bug. Can you create an issue with this information to track it?

**Reply by drolevar** [reply]
Done #6376

**Reply by rseanhall** [reply]
@drolevar We are having problems with GitHub Issues not automatically applying the `triage` label when people create issues, and it happened again with #6376. Do you remember the steps you took to create that issue?

**Reply by drolevar** [reply]
@rseanhall I chose to automatically create an issue from discussion.
![image](https://user-images.githubusercontent.com/14980349/109997785-da478100-7d10-11eb-86f1-40cfe25f112a.png)
